src/config/setshortcutwidget.h

```diff
@@ -14,7 +14,7 @@ class SetShortcutDialog : public QDialog
 {
     Q_OBJECT
 public:
-    explicit SetShortcutDialog(QDialog* parent = nullptr);
+    explicit SetShortcutDialog(QDialog* parent = nullptr, QString shortcutName = "");
```

Comment:
@borgmanJeremy Actually, maybe we could gradually refactor the code to avoid using const refs with Qt containers because they use [implicit sharing](https://doc.qt.io/qt-5/implicit-sharing.html). Saves us from the extra typing and if we decide that the function parameter should be modified, we don't have to make any changes in the function signature, which in turn would makes compilation faster.

Reply:
Interesting, I'll have to think more on this. I like that the type system leads to self documenting code. IE if I pass a const ref its BECAUSE i don't want something to be modified.

Reply:
I personally don't see that `const QString& str` indicates more strongly than `QString str` that `str` won't be modified. For example we don't use `const int&` in parameter declarations (at least I think so). The only difference is that const ref does not perform a deep copy. But because QString objects are implicitly shared, the copy is always shallow unless a deep copy is necessary.

Reply:
Yeah I'm open to either. We can leave this as is and make a larger decision later.